### PR TITLE
feat: add Quorum and Client TLS/SSL encryption for ZooKeeper

### PR DESCRIPTION
**Pull request by marcoppenheimer**
#### `feat: align K8s charm with Machine charm for TLS`
- This is mostly a copy of the recent SSL/TLS implementation for the ZooKeeper Machine charm (https://github.com/canonical/zookeeper-operator/pull/26)
- Some additional minor changes were to be made, including:
    - Tweaking how sans were generated
    - Adding/removing various tests
    - Modifications to the `client` library to have a better interface for SSL connections
    - Modifying the writing/loading of TLS files, and remote running of `keytool` commands on the container
 
## Review Notes
- Focus on L305-L391 of `src/tls.py`, as those needed some updating for K8s.
- Almost all of this code has been reviewed already in https://github.com/canonical/zookeeper-operator/pull/26. As such, very little needs review, but feel free to look through and point out if anything seems odd.

### CODE REVIEW

File: tests/unit/test_provider.py [hunk 1]
```diff
@@ -6,68 +6,58 @@
...
             self.provider.client_relations[0].id, "application", {"chroot": "app"}
         )
         self.harness.update_relation_data(
-            self.provider.app_relation.id, "zookeeper", {"relation-0": "password"}
+            self.provider.app_relation.id, CHARM_KEY, {"relation-0": "password"}
         )
 
-        config = self.harness.charm.client_relation.relation_config(
+        logger.info(self.provider.client_relations[0].data)
+        logger.info(self.provider.app_relation.data)
```
Comment (marceloneppel) [inline tests/unit/test_provider.py:58]: Are these logs only related to debugging during test development and should they be removed?

**Review by marceloneppel — CHANGES_REQUESTED** [review]
Looks great Marc! Only the TLS tests are not configured to run.

File: tests/unit/test_provider.py (hunk 1 above)
Comment (marcoppenheimer) [reply tests/unit/test_provider.py:58]: Very good catch, thank you!!

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -36,4 +83,4 @@ jobs:
           # This is needed until https://bugs.launchpad.net/juju/+bug/1977582 is fixed
           bootstrap-options: "--agent-version 2.9.29"
       - name: Run integration tests
-        run: tox -e integration
+        run: tox -e integration-scaling
```
[the remark this answers is not in the document]
Comment (marcoppenheimer) [reply .github/workflows/ci.yaml:86]: Very good catch, thank you!!

**Review by marceloneppel — APPROVED** [review]
LGTM!

**Pull request merged**